### PR TITLE
T5497: Add ability to resequence rule numbers for firewall

### PR DESCRIPTION
**Pull request by JeffWDH**
## Change Summary
The first draft of the rule generator would sometimes output lines from nodes that did not need re-sequencing (such as set firewall group). This change removes global-options, group and flowtable before getting sent to the re-sequencing function.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T5497

## Component(s) name

generate firewall rule-resequence

## Proposed changes

This change removes global-options, group and flowtable before getting sent to the re-sequencing function.

## How to test
Example of configuration that triggered the bug:  
https://vyos.dev/T5497#160905

The proposed change fixes that. 

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/op_mode/generate_firewall_rule-resequence.py [hunk 1]
```diff
@@ -119,6 +119,11 @@ def convert_rule_keys_to_int(config_dict):
     #config_dict =  config.get_config_dict('firewall')
     config_dict = config.get_config_dict('firewall')
 
+    # Remove global-options, flowtable and group as they don't need sequencing
+    del config_dict['firewall']['global-options']
```
Comment (sever-sever) [inline src/op_mode/generate_firewall_rule-resequence.py]: You should check if those options exist in the dictionary.
For example, I don't have `global-options`
```
vyos@r4:~$ generate firewall rule-resequence 
Traceback (most recent call last):
  File "/usr/libexec/vyos/op_mode/generate_firewall_rule-resequence.py", line 123, in <module>
    del config_dict['firewall']['global-options']
        ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
KeyError: 'global-options'
vyos@r4:~$ 

```

**Review by sever-sever — CHANGES_REQUESTED** [review]

File: src/op_mode/generate_firewall_rule-resequence.py (hunk 1 above)
Comment (JeffWDH) [reply src/op_mode/generate_firewall_rule-resequence.py]: Thanks for catching my newbie mistake :) 

Added proposed change. I don't have flowtable in my config and it doesn't throw an error now. I forgot to test that one after I added it last time.

File: src/op_mode/generate_firewall_rule-resequence.py [hunk 2]
```diff
@@ -116,9 +116,18 @@ def convert_rule_keys_to_int(config_dict):
         print('Firewall is not configured')
         exit(1)
 
-    #config_dict =  config.get_config_dict('firewall')
     config_dict = config.get_config_dict('firewall')
 
+    # Remove global-options, group and flowtable as they don't need sequencing
+    if 'global-options' in config_dict['firewall']:
```
Comment (sarthurdev) [inline src/op_mode/generate_firewall_rule-resequence.py:122]: This (and in the `del` statement) should be be the node mangled value 'global_options' (dashes mangled to underscores)

**Review by sarthurdev — CHANGES_REQUESTED** [review]

**Review by sarthurdev — APPROVED** [review]

**Comment by sever-sever** [comment]
Thanks.
The fix looks good, but there are 5 commits and part of them doesn't include the task number in the commit message.

**Pull request closed without merging**